lighthouse-core/fraggle-rock/user-flow.js

```diff
@@ -108,20 +109,60 @@ class UserFlow {
    */
   async navigate(requestor, stepOptions) {
     if (this.currentTimespan) throw new Error('Timespan already in progress');
+    if (this.currentNavigation) throw new Error('Navigation already in progress');
 
     const options = this._getNextNavigationOptions(stepOptions);
     const gatherResult = await navigationGather(requestor, options);
 
     this._addGatherStep(gatherResult, options);
+  }
+
+  /**
+   * @param {StepOptions=} stepOptions
+   */
+  async startNavigation(stepOptions) {
+    /** @type {ExternalPromise<() => void>} */
+    const setupPromise = new ExternalPromise();
```

Comment:
why can't this just be a promise?  looking at the extpromise impl i'm not seeing what's special about it…

Reply:
There's a big amount of indirection here that takes a moment to figure out. I tried to help in my comment suggestion.

Reply:
The main goal was to make the code easier to read by reducing the nested promises and such. It was a generic pattern that we might user later, but that's just wishful thinking I guess.

I'm fine going with paul's suggestion here.